docs/reference/queries.txt

```diff
@@ -1243,11 +1243,22 @@ Mongoid also has some helpful methods on criteria.
        *Determine if any matching documents exist. Will return true if there
        are 1 or more.*
 
+       ``#exists?`` *now takes a number of argument types:*
+
+       - ``Hash``: *A hash of conditions.*
+       - ``BSON::ObjectId``: *An _id to search for.*
+       - ``String``: *An _id corresponding to the given string to search for.*
+       - ``false``: *Always returns false.*
```

Comment:
Could you explain where the last two usages come from? The `false` one I don't see in https://apidock.com/rails/ActiveRecord/Base/exists%3f/class at all. The string one, in my estimation, comes from AR using integers for ids and providing the convenience shortcut of accepting a string from browser form submission which would be cast to integer.

In MongoDB _id can be of any type, so the second line should be `Object: an _id to search for`. String input I suppose could be handy but I'm not seeing where in this diff this is actually handled, if string input is handled via ObjectId's mongoizer then this documentation here should say that any input which is handled by the defined _id type will be appropriately cast.

Reply:
I originally didn't find the false either... I only saw it in the link that @comandeo sent: https://api.rubyonrails.org/v7.0.3.1/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F

Yes, you're right I will change that about the object id.

Reply:
@p-mongo apidock.com/rails seems to be broken. It shows the documentation for the _Ruby on Rails latest stable (**v5.2.3**)_ However, it is still the very first when one google a rails method :facepalm:

Reply:
I see, that documentation doesn't explain the use case of `false` but since Rails cannot be wrong I approve us copying it.

Reply:
sounds good to me!